presto-product-tests/src/main/java/com/facebook/presto/tests/SqlCancelTests.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tests;
+
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+import com.teradata.tempto.AfterTestWithContext;
+import com.teradata.tempto.BeforeTestWithContext;
+import com.teradata.tempto.ProductTest;
+import com.teradata.tempto.query.QueryExecutionException;
+import io.airlift.http.client.HttpStatus;
+import io.airlift.http.client.Response;
+import io.airlift.log.Logger;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static com.facebook.presto.tests.TestGroups.CANCEL_QUERY;
+import static com.teradata.tempto.assertions.QueryAssert.assertThat;
+import static com.teradata.tempto.query.QueryExecutor.query;
+import static java.lang.String.format;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.fail;
+
+public class SqlCancelTests
+        extends ProductTest
+{
+    private static String tableName = "table_name";
+
+    private static ExecutorService executor;
+
+    @Inject
+    @Named("databases.presto.server_address")
+    private String serverAddress;
+
+    @BeforeTestWithContext
+    @AfterTestWithContext
+    public void dropTestTables()
```

Comment:
Create one method, `beforeTest()`, and annotate it with `@BeforeTestWithContext`. This will contain `executor = Executors.newFixedThreadPool(2);` and a call to `dropTestTables()` (which should not be annotated with anything).

Create another method, `afterTest()`, and annotate it with `@AfterTestWithContext`. Both drop the test tables and close the executor there.